### ISSUE TITLE
Better card declined reporting

### ISSUE THREAD

**Issue by hennaboy** [developermypos/myPOS-SDK-Android#28]
Currently if a card is declined you only see a notification on the go2 reader the app returns a white box followed by a long wait and a message something went wrong 

Surely this should throw a message card declined

**Comment by StiliyanKovachev** [comment]
can you specify your sdk version and what function for begin purchase you are using, POSHandler.purchase or  POSHandler.openPaymentActivity

**Comment by hennaboy** [comment]
SDK was downloaded from here https://f.mypos.com/index.php/s/SKHWLCiPptNd9d6

Version 2.1.3

PosHandler.oprnPaymentActivity()

**Comment by StiliyanKovachev** [comment]
please download latest version 2.1.6

https://github.com/developermypos/myPOS-SDK-Android/tree/master/slavesdk

if the issue still there, please provide a steps for reproducing, more details about the declined code and also provide TID and send log from the terminal

**Comment by hennaboy** [comment]
Tested this morning and is much better and the response is very quick now but the response text is operation not permitted whereas surely it would be better for the response to be Card Declined.

**Comment by hennaboy** [comment]
Also, insert card if contactless is not accepted. Should return a response to the screen but it does not therefore if you have the card reader facing away from you towards the customer you have to look to see if that is what is occurring as there is no notification that is what is required.

Could it not return a response to the SDK screen Insert Card?

**Comment by StiliyanKovachev** [comment]
Those are issues related to the POS device, Please send this requests to online@mypos.com . There the developers support team can fully support you.

**Comment by hennaboy** [comment]
> Those are issues related to the POS device, Please send this requests to [online@mypos.com](mailto:online@mypos.com) . There the developers support team can fully support you.

How is it related to the POS device?

The text Operation Not Permitted is coming from the SDK language file is it not?

**Comment by StiliyanKovachev** [comment]
yes and it is related to specific status from the IPProtocol that sdk and POS device use to communicate
